internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -1023,12 +1023,266 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Valid integer schema with CEL expression",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "cpu",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "integer",
+						XValidations: []clusterv1.ValidationRule{{
+							Rule: "self >= 1",
+						}},
+					},
+				},
+			},
+		},
+		{
+			name: "Integer schema with invalid CEL expression",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "cpu",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "integer",
+						XValidations: []clusterv1.ValidationRule{{
+							Rule: "this does not compile",
+						}},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Forbid validation rules where cost total exceeds total limit",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "cpu",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"list": {
+								Type:     "array",
+								MaxItems: ptr.To[int64](10000000),
+								Items: &clusterv1.JSONSchemaProps{
+									Type:         "string",
+									MaxLength:    ptr.To[int64](500000),
+									XValidations: []clusterv1.ValidationRule{{Rule: "self.contains('keyword')"}},
+								},
+							},
+							"field": { // include a validation rule that does not contribute to total limit being exceeded (i.e. it is less than 1% of the limit)
+								Type:         "integer",
+								XValidations: []clusterv1.ValidationRule{{Rule: "self > 50 && self < 100"}},
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "x-kubernetes-validations should have valid reason and fieldPath",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						XValidations: []clusterv1.ValidationRule{
+							{
+								Rule:      "self.a > 0",
+								Reason:    clusterv1.FieldValueErrorReason("InternalError"),
+								FieldPath: ".a",
+							},
+						},
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"a": {
+								Type: "number",
+								XValidations: []clusterv1.ValidationRule{
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueRequired,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueInvalid,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueDuplicate,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueForbidden,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "x-kubernetes-validations should have valid fieldPath for array",
```

Comment:
Let's please add a comment inline which one is the invalid field

Reply:
Updated to add better checks for failure errors to make this clear and make the test more valuable.